src/components/Footer.tsx

```diff
@@ -0,0 +1,105 @@
+import cn from 'classnames';
+import { useContext } from 'react';
+
+import { FilterStatus } from '../types/FilterStatus';
+import { DispatchContext, StateContext } from './TodosProvider';
+import { deleteTodo } from '../api/todos';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { LoadingStatus } from '../types/LoadingStatus';
+
+export const Footer = () => {
+  const { filteredBy, todos, newTodoInputRef } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const filteredStatus = Object.values(FilterStatus);
+
+  const completedTodosCount = todos.filter(
```

Comment:
wrap those operation where you use filter or find (or sth like this) with useMemo hook